src/analysis/processing/qgsalgorithmmultidifference.cpp

```diff
@@ -0,0 +1,176 @@
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ) ) );
+  addParameter( new QgsProcessingParameterMultipleLayers( QStringLiteral( "OVERLAYS" ), QObject::tr( "Overlay layers" ), QgsProcessing::TypeVector ) );
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Difference" ) ) );
+}
+
+
+QVariantMap QgsMultiDifferenceAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  std::unique_ptr< QgsFeatureSource > sourceA( parameterAsSource( parameters, QStringLiteral( "INPUT" ), context ) );
+  if ( !sourceA )
+    throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "INPUT" ) ) );
+
+  const QList< QgsMapLayer * > layers = parameterAsLayerList( parameters, QStringLiteral( "OVERLAYS" ), context );
+
+  // loop through overlay layers and check whether they are vectors
+  long totalLayerCount = 0;
+  for ( QgsMapLayer *layer : layers )
+  {
+    if ( feedback->isCanceled() )
+      break;
+
+    if ( !layer )
+      throw QgsProcessingException( QObject::tr( "Error retrieving map layer." ) );
+
+    if ( layer->type() != QgsMapLayerType::VectorLayer )
+      throw QgsProcessingException( QObject::tr( "All layers must be vector layers!" ) );
+
+    totalLayerCount++;
+  }
+
+  const QgsWkbTypes::Type geometryType = QgsWkbTypes::multiType( sourceA->wkbType() );
+  const QgsCoordinateReferenceSystem crs = sourceA->sourceCrs();
+  std::unique_ptr< QgsFeatureSink > sink;
+  long count = 0;
+  QVariantMap outputs;
+
+  if ( totalLayerCount == 1 )
+  {
+    QString dest;
+    sink.reset( parameterAsSink( parameters, QStringLiteral( "OUTPUT" ), context, dest, sourceA->fields(), geometryType, crs ) );
+    if ( !sink )
+      throw QgsProcessingException( invalidSinkError( parameters, QStringLiteral( "OUTPUT" ) ) );
+
+    outputs.insert( QStringLiteral( "OUTPUT" ), dest );
+
+    QgsVectorLayer *overlayLayer = qobject_cast< QgsVectorLayer * >( layers.at( 0 ) );
+
+    const long total = sourceA->featureCount();
+    QgsOverlayUtils::difference( *sourceA, *overlayLayer, *sink, context, feedback, count, total, QgsOverlayUtils::OutputA );
+  }
+  else
+  {
+    QgsProcessingMultiStepFeedback multiStepFeedback( totalLayerCount, feedback );
+
+    QString id = QStringLiteral( "memory:" );
+    QgsVectorLayer *differenceLayer;
+
+    long i = 0;
+    for ( QgsMapLayer *layer : layers )
+    {
+      if ( feedback->isCanceled() )
+        break;
+
+      multiStepFeedback.setCurrentStep( i );
+
+      if ( !layer )
+        continue;
+
+      QgsVectorLayer *overlayLayer = qobject_cast< QgsVectorLayer * >( layer );
+      if ( !overlayLayer )
+        continue;
+
+      count = 0;
+      if ( i == 0 )
+      {
+        sink.reset( QgsProcessingUtils::createFeatureSink( id, context, sourceA->fields(), geometryType, crs ) );
```

Comment:
```suggestion
        sink.reset( QgsProcessingUtils::createFeatureSink( QStringLiteral("memory:"), context, sourceA->fields(), geometryType, crs ) );
```

i'd move this down here -- it makes the code slightly more readable (same with line 163)

Reply:
I see your point, but later in line 147 we have to get layer from the sink id, so it is necessary to keep it in the variable for further use.